src/EIP712.sol

```diff
@@ -47,6 +48,23 @@ contract EIP712 is IEIP712, IERC5267 {
             bytes32 salt,
             uint256[] memory extensions
         )
+    {
+        (fields, name, version, chainId, verifyingContract, salt, extensions) = _eip712Domain();
+    }
+
+    /// @notice Internal eip712Domain implementation for use in ERC7739
+    function _eip712Domain()
```

Comment:
why not just make public?

Reply:
oh I thought internal funcs have gas savings